Impossible to compile examples

### Description

I'm trying to recompile ESPUI-gui test but it fails. It seems that some methods: ```canSend()``` and ```redirect()``` are not implemented in ``` emuESPAsyncWebServer```

### Steps to Reproduce

1. Installed necessary libraries via arduino-cli, cloned [ESP8266/Arduino](https://github.com/esp8266/Arduino):

```bash
arduino-cli core update-index
curl https://raw.githubusercontent.com/espressif/arduino-esp32/gh-pages/package_esp32_index.json --output .arduino15/package_esp32_index.json
arduino-cli core install -v esp32:esp32 --additional-urls https://raw.githubusercontent.com/espressif/arduino-esp32/gh-pages/package_esp32_index.json
arduino-cli config init
arduino-cli config set library.enable_unsafe_install true
arduino-cli lib install -v ArduinoJson
arduino-cli lib install -v ESPAsyncWebServer
arduino-cli lib install -v AsyncTCP
arduino-cli lib install -v ESPUI
arduino-cli lib install -v --git-url https://github.com/Links2004/arduinoWebSockets

[vagrant@rhel8 tests]$ arduino-cli lib list
Name                Installed Available Location              Description
ArduinoJson         6.20.0    -         LIBRARY_LOCATION_USER -          
emuAsync                      -         LIBRARY_LOCATION_USER -          
ESP Async WebServer 1.2.3     -         LIBRARY_LOCATION_USER -          
ESP AsyncTCP        1.2.2     -         LIBRARY_LOCATION_USER -          
ESPUI               2.2.1     -         LIBRARY_LOCATION_USER -          
LittleFS_esp32      1.0.6     -         LIBRARY_LOCATION_USER -          
WebSockets          2.3.7     -         LIBRARY_LOCATION_USER -          

[vagrant@rhel8 tests]$ arduino-cli version
arduino-cli  Version: 0.29.0 Commit: 76251df9 Date: 2022-11-17T09:21:40Z
```
2. Trying to compile the example:

```bash
[vagrant@rhel8 libraries]$ cd emuAsync/tests/
[vagrant@rhel8 tests]$ pwd
/home/vagrant/Arduino/libraries/emuAsync/tests
[vagrant@rhel8 tests]$ ./ESPUI-gui
```

**Expected behavior:** 
To be able to run ESPUI-gui in browser

**Actual behavior:**

```bash
+ cd /home/vagrant/eps8266_arduino/tests/host
+ make FORCE32=0 ssl
Makefile:21: using /usr/bin/g++ and /usr/bin/gcc
Makefile:47: compiling in native mode
cd ../../tools/sdk/ssl && make native
make[1]: Entering directory '/home/vagrant/eps8266_arduino/tools/sdk/ssl'
cd bearssl && make
make[2]: Entering directory '/home/vagrant/eps8266_arduino/tools/sdk/ssl/bearssl'
make[2]: Nothing to be done for 'all'.
make[2]: Leaving directory '/home/vagrant/eps8266_arduino/tools/sdk/ssl/bearssl'
make[1]: Leaving directory '/home/vagrant/eps8266_arduino/tools/sdk/ssl'
+ make -j 10 FORCE32=0 D=1 'USERCFLAGS=-I /home/vagrant/Arduino/libraries/emuAsync/replacement' ULIBDIRS=/home/vagrant/Arduino/libraries/emuAsync:/home/vagrant/Arduino/libraries/ESPUI:/home/vagrant/Arduino/libraries/ArduinoJson:/home/vagrant/Arduino/libraries/arduinoWebSockets /home/vagrant/Arduino/libraries/ESPUI/examples/gui/gui
Makefile:21: using /usr/bin/g++ and /usr/bin/gcc
Makefile:47: compiling in native mode
make INO=/home/vagrant/Arduino/libraries/ESPUI/examples/gui/gui.ino /home/vagrant/eps8266_arduino/tests/host/bin//home/vagrant/Arduino/libraries/ESPUI/examples/gui/gui
make[1]: Entering directory '/home/vagrant/eps8266_arduino/tests/host'
Makefile:21: using /usr/bin/g++ and /usr/bin/gcc
Makefile:47: compiling in native mode
C++ /home/vagrant/eps8266_arduino/tests/host/bin//home/vagrant/Arduino/libraries/ESPUI/examples/gui/gui.ino.cpp.o
C++ /home/vagrant/eps8266_arduino/tests/host/bin//home/vagrant/Arduino/libraries/ESPUI/src/ESPUIclient.cpp.o
C++ /home/vagrant/eps8266_arduino/tests/host/bin//home/vagrant/Arduino/libraries/ESPUI/src/ESPUI.cpp.o
/home/vagrant/Arduino/libraries/ESPUI/src/ESPUIclient.cpp: In member function 'bool ESPUIclient::CanSend()':
/home/vagrant/Arduino/libraries/ESPUI/src/ESPUIclient.cpp:36:28: error: 'class AsyncWebSocketClient' has no member named 'canSend'
         Response = client->canSend();
                            ^~~~~~~
make[1]: *** [Makefile:288: /home/vagrant/eps8266_arduino/tests/host/bin//home/vagrant/Arduino/libraries/ESPUI/src/ESPUIclient.cpp.o] Error 1
make[1]: *** Waiting for unfinished jobs....
In file included from /home/vagrant/eps8266_arduino/tests/host/bin//home/vagrant/Arduino/libraries/ESPUI/examples/gui/gui.ino.cpp:1:
/home/vagrant/Arduino/libraries/ESPUI/examples/gui//gui.ino: In function 'void buttonExample(Control*, int, void*)':
/home/vagrant/Arduino/libraries/ESPUI/examples/gui//gui.ino:65:56: error: cast from 'void*' to 'int' loses precision [-fpermissive]
     Serial.println(String("param: ") + String(int(param)));
                                                        ^
make[1]: *** [Makefile:291: /home/vagrant/eps8266_arduino/tests/host/bin//home/vagrant/Arduino/libraries/ESPUI/examples/gui/gui.ino.cpp.o] Error 1
/home/vagrant/Arduino/libraries/ESPUI/src/ESPUI.cpp: In lambda function:
/home/vagrant/Arduino/libraries/ESPUI/src/ESPUI.cpp:1250:13: error: 'using AsyncWebServerRequest = class AsyncWebServer' {aka 'class AsyncWebServer'} has no member named 'redirect'
    request->redirect("/");
             ^~~~~~~~
/home/vagrant/Arduino/libraries/ESPUI/src/ESPUI.cpp: In lambda function:
/home/vagrant/Arduino/libraries/ESPUI/src/ESPUI.cpp:1408:13: error: 'using AsyncWebServerRequest = class AsyncWebServer' {aka 'class AsyncWebServer'} has no member named 'redirect'
    request->redirect("/");
             ^~~~~~~~
make[1]: *** [Makefile:288: /home/vagrant/eps8266_arduino/tests/host/bin//home/vagrant/Arduino/libraries/ESPUI/src/ESPUI.cpp.o] Error 1
make[1]: Leaving directory '/home/vagrant/eps8266_arduino/tests/host'
make: *** [Makefile:396: /home/vagrant/Arduino/libraries/ESPUI/examples/gui/gui] Error 2
```

### Versions
OS: RHEL 8.7

Guys, can you add missing methods?  Thanks a lot!
Pavol

--- a/emuESPAsyncWebServer.h
+++ b/emuESPAsyncWebServer.h
@@ -206,6 +206,12 @@ public:
         return nullptr;
     }
 
+    void redirect (const String& url)
+    {
+        sendHeader("Location", url, true);
+        send(302);
+    }
+
 protected:
 
     void checkResponse ()
@@ -243,6 +249,7 @@ public:
 
     uint32_t id () const { return _id; }
     void text(const String& t);
+    bool canSend () { return true; }
 
 protected:
     AsyncWebSocket* _srv = nullptr;

